tests/phpunit/integration/Core/Util/REST_Entity_Search_ControllerTest.php

```diff
@@ -0,0 +1,172 @@
+	}
+
+	public function test_register() {
+		remove_all_filters( 'googlesitekit_rest_routes' );
+
+		$this->controller->register();
+
+		$this->assertTrue( has_filter( 'googlesitekit_rest_routes' ) );
+	}
+
+	public function test_unauthorized_request() {
+		$this->controller->register();
+
+		$request = new WP_REST_Request( WP_REST_Server::READABLE, '/' . REST_Routes::REST_ROOT . '/core/search/data/entity-search' );
+		$request->set_query_params(
+			array(
+				'query' => 'foo',
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertNotEquals( 200, $response->get_status() );
+		$this->assertArrayHasKey( 'code', $response->get_data() );
+		$this->assertEquals( 'rest_forbidden', $response->get_data()['code'] );
+	}
+
+	public function test_authorized_request() {
+		$this->controller->register();
+
+		$user_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user_id );
+
+		$request = new WP_REST_Request( WP_REST_Server::READABLE, '/' . REST_Routes::REST_ROOT . '/core/search/data/entity-search' );
+		$request->set_query_params(
+			array(
+				'query' => 'foo',
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertEquals( 200, $response->get_status() );
+	}
+
+	public function test_authorized_request_without_query() {
+		$this->controller->register();
+
+		$user_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user_id );
+
+		$request  = new WP_REST_Request( WP_REST_Server::READABLE, '/' . REST_Routes::REST_ROOT . '/core/search/data/entity-search' );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertEquals( 400, $response->get_status() );
+		$this->assertArrayHasKey( 'code', $response->get_data() );
+		$this->assertEquals( 'rest_missing_callback_param', $response->get_data()['code'] );
+	}
+
+	public function test_post_search() {
+		$this->controller->register();
+
+		$user_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $user_id );
+
+		$this->factory()->post->create(
+			array(
+				'post_status' => 'publish',
+				'post_title'  => 'Hello Earth',
+			)
+		);
+
+		$this->factory()->post->create(
+			array(
+				'post_status' => 'publish',
+				'post_title'  => 'Hello Mars',
+			)
+		);
+
+		$this->factory()->post->create(
+			array(
+				'post_status' => 'publish',
+				'post_title'  => 'Howdy Jupiter',
+			)
+		);
+
+		$request = new WP_REST_Request( WP_REST_Server::READABLE, '/' . REST_Routes::REST_ROOT . '/core/search/data/entity-search' );
+		$request->set_query_params(
+			array(
+				'query' => 'hello',
+			)
+		);
+
+		$response = rest_get_server()->dispatch( $request );
+		$data     = $response->get_data();
+
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( 2, count( $data ) );
+		$this->assertEquals( 'post', $data[0]['type'] );
+		$this->assertArrayHasKey( 'id', $data[0] );
```

Comment:
Hey @kuasha420, thanks for making this change. However, this test is still not asserting that the correct posts have been returned. Could you please either take an approach similar to the `test_entity_search` where you verify the ID matches, or otherwise test on the title?